api/python/provisioner/commands/validator/validator.py

```diff
@@ -784,3 +787,128 @@ def __attrs_post_init__(self):
         self.content_validator = ContentFileValidator(
                 scheme=ThirdPartyReleaseInfoContentScheme,
                 content_type=self.content_type)
+
+
+@attr.s(auto_attribs=True)
+class CompatibilityValidator:
+
+    """
+    Validator that checks if all SW upgrade packages are compatible with
+    installed ones.
+    """
+
+    @staticmethod
+    def _compare_versions(a: str, b: str, operator: str) -> bool:
+        """
+        Compare two version using packaging.version.parse method
+
+        Parameters
+        ----------
+        a: str
+            The first operand
+        b: str
+            The second operand
+        operator: str
+            String representation of the operator
+
+        Returns
+        -------
+        bool:
+            return True if comparison is true and False otherwise
+
+        """
+        operator = ComparisonOperators(operator)
+
+        if operator == ComparisonOperators.GREATER:
+            return version.parse(a) > version.parse(b)
+        elif operator == ComparisonOperators.LOWER:
+            return version.parse(a) < version.parse(b)
+        elif operator == ComparisonOperators.EQUAL:
+            return version.parse(a) == version.parse(b)
+        elif operator == ComparisonOperators.NOT_EQUAL:
+            return version.parse(a) != version.parse(b)
+        elif operator == ComparisonOperators.GREATER_OR_EQUAL:
+            return version.parse(a) >= version.parse(b)
+        elif operator == ComparisonOperators.LOWER_OR_EQUAL:
+            return version.parse(a) <= version.parse(b)
+        else:
+            raise ValidationError(
+                f"Unexpected comparison operator: '{operator}'")
+
+    def validate(self, iso_info):
+        """
+
+        Parameters
+        ----------
+        iso_info: CortxISOInfo
+            CortxISOInfo instance with all necessary information about
+            SW upgrade ISO
+
+        Returns
+        -------
+        None
+
+        Raises
+        ------
+        ValidationError
+            If validation is failed.
+        """
+        packages = list(iso_info.packages.keys())
+
+        # NOTE: the first line of `yum -q list installed` command is
+        #  'Installed Packages' skip it via `tail -n +2`
+        cmd = (f"yum -q list installed {packages} | tail -n +2 | "
```

Comment:
edge case: packages list is empty

Reply:
It is not a problem, but it is better to stop validation in this case to save the CPU time. Done